# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/libGLESv2/libGLESv2_autogen.cpp
# Commit: 13282d7a47a5
# Full Hash: 13282d7a47a57a73be8d060efd9645b4d642f03e
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-09-08 09:46:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 3064469c073d -r 13282d7a47a5 gfx/angle/checkout/src/libGLESv2/libGLESv2_autogen.cpp
--- a/gfx/angle/checkout/src/libGLESv2/libGLESv2_autogen.cpp	Sat Sep 07 13:36:46 2019 +0000
+++ b/gfx/angle/checkout/src/libGLESv2/libGLESv2_autogen.cpp	Sat Sep 07 13:38:36 2019 +0000
@@ -3197,6 +3197,25 @@
     return gl::GetQueryObjectui64vRobustANGLE(id, pname, bufSize, length, params);
 }
 
+// GL_ANGLE_texture_external_update
+void GL_APIENTRY glTexImage2DExternalANGLE(GLenum target,
+                                           GLint level,
+                                           GLint internalformat,
+                                           GLsizei width,
+                                           GLsizei height,
+                                           GLint border,
+                                           GLenum format,
+                                           GLenum type)
+{
+    return gl::TexImage2DExternalANGLE(target, level, internalformat, width, height, border, format,
+                                       type);
+}
+
+void GL_APIENTRY glInvalidateTextureANGLE(GLenum target)
+{
+    return gl::InvalidateTextureANGLE(target);
+}
+
 // GL_ANGLE_texture_multisample
 void GL_APIENTRY glTexStorage2DMultisampleANGLE(GLenum target,
                                                 GLsizei samples,
@@ -3309,6 +3328,12 @@
     return gl::MatrixLoadIdentityCHROMIUM(matrixMode);
 }
 
+// GL_CHROMIUM_lose_context
+void GL_APIENTRY glLoseContextCHROMIUM(GLenum current, GLenum other)
+{
+    return gl::LoseContextCHROMIUM(current, other);
+}
+
 // GL_CHROMIUM_path_rendering
 GLuint GL_APIENTRY glGenPathsCHROMIUM(GLsizei range)
 {
@@ -3650,6 +3675,109 @@
     return gl::MapBufferRangeEXT(target, offset, length, access);
 }
 
+// GL_EXT_memory_object
+void GL_APIENTRY glBufferStorageMemEXT(GLenum target,
+                                       GLsizeiptr size,
+                                       GLuint memory,
+                                       GLuint64 offset)
+{
+    return gl::BufferStorageMemEXT(target, size, memory, offset);
+}
+
+void GL_APIENTRY glCreateMemoryObjectsEXT(GLsizei n, GLuint *memoryObjects)
+{
+    return gl::CreateMemoryObjectsEXT(n, memoryObjects);
+}
+
+void GL_APIENTRY glDeleteMemoryObjectsEXT(GLsizei n, const GLuint *memoryObjects)
+{
+    return gl::DeleteMemoryObjectsEXT(n, memoryObjects);
+}
+
+void GL_APIENTRY glGetMemoryObjectParameterivEXT(GLuint memoryObject, GLenum pname, GLint *params)
+{
+    return gl::GetMemoryObjectParameterivEXT(memoryObject, pname, params);
+}
+
+void GL_APIENTRY glGetUnsignedBytevEXT(GLenum pname, GLubyte *data)
+{
+    return gl::GetUnsignedBytevEXT(pname, data);
+}
+
+void GL_APIENTRY glGetUnsignedBytei_vEXT(GLenum target, GLuint index, GLubyte *data)
+{
+    return gl::GetUnsignedBytei_vEXT(target, index, data);
+}
+
+GLboolean GL_APIENTRY glIsMemoryObjectEXT(GLuint memoryObject)
+{
+    return gl::IsMemoryObjectEXT(memoryObject);
+}
+
+void GL_APIENTRY glMemoryObjectParameterivEXT(GLuint memoryObject,
+                                              GLenum pname,
+                                              const GLint *params)
+{
+    return gl::MemoryObjectParameterivEXT(memoryObject, pname, params);
+}
+
+void GL_APIENTRY glTexStorageMem2DEXT(GLenum target,
+                                      GLsizei levels,
+                                      GLenum internalFormat,
+                                      GLsizei width,
+                                      GLsizei height,
+                                      GLuint memory,
+                                      GLuint64 offset)
+{
+    return gl::TexStorageMem2DEXT(target, levels, internalFormat, width, height, memory, offset);
+}
+
+void GL_APIENTRY glTexStorageMem2DMultisampleEXT(GLenum target,
+                                                 GLsizei samples,
+                                                 GLenum internalFormat,
+                                                 GLsizei width,
+                                                 GLsizei height,
+                                                 GLboolean fixedSampleLocations,
+                                                 GLuint memory,
+                                                 GLuint64 offset)
+{
+    return gl::TexStorageMem2DMultisampleEXT(target, samples, internalFormat, width, height,
+                                             fixedSampleLocations, memory, offset);
+}
+
+void GL_APIENTRY glTexStorageMem3DEXT(GLenum target,
+                                      GLsizei levels,
+                                      GLenum internalFormat,
+                                      GLsizei width,
+                                      GLsizei height,
+                                      GLsizei depth,
+                                      GLuint memory,
+                                      GLuint64 offset)
+{
+    return gl::TexStorageMem3DEXT(target, levels, internalFormat, width, height, depth, memory,
+                                  offset);
+}
+
+void GL_APIENTRY glTexStorageMem3DMultisampleEXT(GLenum target,
+                                                 GLsizei samples,
+                                                 GLenum internalFormat,
+                                                 GLsizei width,
+                                                 GLsizei height,
+                                                 GLsizei depth,
+                                                 GLboolean fixedSampleLocations,
+                                                 GLuint memory,
+                                                 GLuint64 offset)
+{
+    return gl::TexStorageMem3DMultisampleEXT(target, samples, internalFormat, width, height, depth,
+                                             fixedSampleLocations, memory, offset);
+}
+
+// GL_EXT_memory_object_fd
+void GL_APIENTRY glImportMemoryFdEXT(GLuint memory, GLuint64 size, GLenum handleType, GLint fd)
+{
+    return gl::ImportMemoryFdEXT(memory, size, handleType, fd);
+}
+
 // GL_EXT_occlusion_query_boolean
 
 // GL_EXT_robustness
@@ -3683,6 +3811,62 @@
     return gl::ReadnPixelsEXT(x, y, width, height, format, type, bufSize, data);
 }
 
+// GL_EXT_semaphore
+void GL_APIENTRY glDeleteSemaphoresEXT(GLsizei n, const GLuint *semaphores)
+{
+    return gl::DeleteSemaphoresEXT(n, semaphores);
+}
+
+void GL_APIENTRY glGenSemaphoresEXT(GLsizei n, GLuint *semaphores)
+{
+    return gl::GenSemaphoresEXT(n, semaphores);
+}
+
+void GL_APIENTRY glGetSemaphoreParameterui64vEXT(GLuint semaphore, GLenum pname, GLuint64 *params)
+{
+    return gl::GetSemaphoreParameterui64vEXT(semaphore, pname, params);
+}
+
+GLboolean GL_APIENTRY glIsSemaphoreEXT(GLuint semaphore)
+{
+    return gl::IsSemaphoreEXT(semaphore);
+}
+
+void GL_APIENTRY glSemaphoreParameterui64vEXT(GLuint semaphore,
+                                              GLenum pname,
+                                              const GLuint64 *params)
+{
+    return gl::SemaphoreParameterui64vEXT(semaphore, pname, params);
+}
+
+void GL_APIENTRY glSignalSemaphoreEXT(GLuint semaphore,
+                                      GLuint numBufferBarriers,
+                                      const GLuint *buffers,
+                                      GLuint numTextureBarriers,
+                                      const GLuint *textures,
+                                      const GLenum *dstLayouts)
+{
+    return gl::SignalSemaphoreEXT(semaphore, numBufferBarriers, buffers, numTextureBarriers,
+                                  textures, dstLayouts);
+}
+
+void GL_APIENTRY glWaitSemaphoreEXT(GLuint semaphore,
+                                    GLuint numBufferBarriers,
+                                    const GLuint *buffers,
+                                    GLuint numTextureBarriers,
+                                    const GLuint *textures,
+                                    const GLenum *srcLayouts)
+{
+    return gl::WaitSemaphoreEXT(semaphore, numBufferBarriers, buffers, numTextureBarriers, textures,
+                                srcLayouts);
+}
+
+// GL_EXT_semaphore_fd
+void GL_APIENTRY glImportSemaphoreFdEXT(GLuint semaphore, GLenum handleType, GLint fd)
+{
+    return gl::ImportSemaphoreFdEXT(semaphore, handleType, fd);
+}
+
 // GL_EXT_texture_storage
 void GL_APIENTRY glTexStorage1DEXT(GLenum target,
                                    GLsizei levels,
@@ -4048,6 +4232,91 @@
     return gl::QueryMatrixxOES(mantissa, exponent);
 }
 
+// GL_OES_texture_3D
+void GL_APIENTRY glCompressedTexImage3DOES(GLenum target,
+                                           GLint level,
+                                           GLenum internalformat,
+                                           GLsizei width,
+                                           GLsizei height,
+                                           GLsizei depth,
+                                           GLint border,
+                                           GLsizei imageSize,
+                                           const void *data)
+{
+    return gl::CompressedTexImage3DOES(target, level, internalformat, width, height, depth, border,
+                                       imageSize, data);
+}
+
+void GL_APIENTRY glCompressedTexSubImage3DOES(GLenum target,
+                                              GLint level,
+                                              GLint xoffset,
+                                              GLint yoffset,
+                                              GLint zoffset,
+                                              GLsizei width,
+                                              GLsizei height,
+                                              GLsizei depth,
+                                              GLenum format,
+                                              GLsizei imageSize,
+                                              const void *data)
+{
+    return gl::CompressedTexSubImage3DOES(target, level, xoffset, yoffset, zoffset, width, height,
+                                          depth, format, imageSize, data);
+}
+
+void GL_APIENTRY glCopyTexSubImage3DOES(GLenum target,
+                                        GLint level,
+                                        GLint xoffset,
+                                        GLint yoffset,
+                                        GLint zoffset,
+                                        GLint x,
+                                        GLint y,
+                                        GLsizei width,
+                                        GLsizei height)
+{
+    return gl::CopyTexSubImage3DOES(target, level, xoffset, yoffset, zoffset, x, y, width, height);
+}
+
+void GL_APIENTRY glFramebufferTexture3DOES(GLenum target,
+                                           GLenum attachment,
+                                           GLenum textarget,
+                                           GLuint texture,
+                                           GLint level,
+                                           GLint zoffset)
+{
+    return gl::FramebufferTexture3DOES(target, attachment, textarget, texture, level, zoffset);
+}
+
+void GL_APIENTRY glTexImage3DOES(GLenum target,
+                                 GLint level,
+                                 GLenum internalformat,
+                                 GLsizei width,
+                                 GLsizei height,
+                                 GLsizei depth,
+                                 GLint border,
+                                 GLenum format,
+                                 GLenum type,
+                                 const void *pixels)
+{
+    return gl::TexImage3DOES(target, level, internalformat, width, height, depth, border, format,
+                             type, pixels);
+}
+
+void GL_APIENTRY glTexSubImage3DOES(GLenum target,
+                                    GLint level,
+                                    GLint xoffset,
+                                    GLint yoffset,
+                                    GLint zoffset,
+                                    GLsizei width,
+                                    GLsizei height,
+                                    GLsizei depth,
+                                    GLenum format,
+                                    GLenum type,
+                                    const void *pixels)
+{
+    return gl::TexSubImage3DOES(target, level, xoffset, yoffset, zoffset, width, height, depth,
+                                format, type, pixels);
+}
+
 // GL_OES_texture_border_clamp
 void GL_APIENTRY glGetSamplerParameterIivOES(GLuint sampler, GLenum pname, GLint *params)
 {
@@ -4426,6 +4695,15 @@
     return gl::BufferDataContextANGLE(ctx, target, size, data, usage);
 }
 
+void GL_APIENTRY glBufferStorageMemEXTContextANGLE(GLeglContext ctx,
+                                                   GLenum target,
+                                                   GLsizeiptr size,
+                                                   GLuint memory,
+                                                   GLuint64 offset)
+{
+    return gl::BufferStorageMemEXTContextANGLE(ctx, target, size, memory, offset);
+}
+
 void GL_APIENTRY glBufferSubDataContextANGLE(GLeglContext ctx,
                                              GLenum target,
                                              GLintptr offset,
@@ -4603,6 +4881,21 @@
                                                 depth, border, imageSize, data);
 }
 
+void GL_APIENTRY glCompressedTexImage3DOESContextANGLE(GLeglContext ctx,
+                                                       GLenum target,
+                                                       GLint level,
+                                                       GLenum internalformat,
+                                                       GLsizei width,
+                                                       GLsizei height,
+                                                       GLsizei depth,
+                                                       GLint border,
+                                                       GLsizei imageSize,
+                                                       const void *data)
+{
+    return gl::CompressedTexImage3DOESContextANGLE(ctx, target, level, internalformat, width,
+                                                   height, depth, border, imageSize, data);
+}
+
 void GL_APIENTRY glCompressedTexSubImage2DContextANGLE(GLeglContext ctx,
                                                        GLenum target,
                                                        GLint level,
@@ -4635,6 +4928,24 @@
                                                    width, height, depth, format, imageSize, data);
 }
 
+void GL_APIENTRY glCompressedTexSubImage3DOESContextANGLE(GLeglContext ctx,
+                                                          GLenum target,
+                                                          GLint level,
+                                                          GLint xoffset,
+                                                          GLint yoffset,
+                                                          GLint zoffset,
+                                                          GLsizei width,
+                                                          GLsizei height,
+                                                          GLsizei depth,
+                                                          GLenum format,
+                                                          GLsizei imageSize,
+                                                          const void *data)
+{
+    return gl::CompressedTexSubImage3DOESContextANGLE(ctx, target, level, xoffset, yoffset, zoffset,
+                                                      width, height, depth, format, imageSize,
+                                                      data);
+}
+
 void GL_APIENTRY glCopyBufferSubDataContextANGLE(GLeglContext ctx,
                                                  GLenum readTarget,
                                                  GLenum writeTarget,
@@ -4689,6 +5000,28 @@
                                              width, height);
 }
 
+void GL_APIENTRY glCopyTexSubImage3DOESContextANGLE(GLeglContext ctx,
+                                                    GLenum target,
+                                                    GLint level,
+                                                    GLint xoffset,
+                                                    GLint yoffset,
+                                                    GLint zoffset,
+                                                    GLint x,
+                                                    GLint y,
+                                                    GLsizei width,
+                                                    GLsizei height)
+{
+    return gl::CopyTexSubImage3DOESContextANGLE(ctx, target, level, xoffset, yoffset, zoffset, x, y,
+                                                width, height);
+}
+
+void GL_APIENTRY glCreateMemoryObjectsEXTContextANGLE(GLeglContext ctx,
+                                                      GLsizei n,
+                                                      GLuint *memoryObjects)
+{
+    return gl::CreateMemoryObjectsEXTContextANGLE(ctx, n, memoryObjects);
+}
+
 GLuint GL_APIENTRY glCreateProgramContextANGLE(GLeglContext ctx)
 {
     return gl::CreateProgramContextANGLE(ctx);
@@ -4770,6 +5103,13 @@
     return gl::DeleteFramebuffersOESContextANGLE(ctx, n, framebuffers);
 }
 
+void GL_APIENTRY glDeleteMemoryObjectsEXTContextANGLE(GLeglContext ctx,
+                                                      GLsizei n,
+                                                      const GLuint *memoryObjects)
+{
+    return gl::DeleteMemoryObjectsEXTContextANGLE(ctx, n, memoryObjects);
+}
+
 void GL_APIENTRY glDeleteProgramContextANGLE(GLeglContext ctx, GLuint program)
 {
     return gl::DeleteProgramContextANGLE(ctx, program);
@@ -4813,6 +5153,13 @@
     return gl::DeleteSamplersContextANGLE(ctx, count, samplers);
 }
 
+void GL_APIENTRY glDeleteSemaphoresEXTContextANGLE(GLeglContext ctx,
+                                                   GLsizei n,
+                                                   const GLuint *semaphores)
+{
+    return gl::DeleteSemaphoresEXTContextANGLE(ctx, n, semaphores);
+}
+
 void GL_APIENTRY glDeleteShaderContextANGLE(GLeglContext ctx, GLuint shader)
 {
     return gl::DeleteShaderContextANGLE(ctx, shader);
@@ -5220,6 +5567,18 @@
                                                    level);
 }
 
+void GL_APIENTRY glFramebufferTexture3DOESContextANGLE(GLeglContext ctx,
+                                                       GLenum target,
+                                                       GLenum attachment,
+                                                       GLenum textarget,
+                                                       GLuint texture,
+                                                       GLint level,
+                                                       GLint zoffset)
+{
+    return gl::FramebufferTexture3DOESContextANGLE(ctx, target, attachment, textarget, texture,
+                                                   level, zoffset);
+}
+
 void GL_APIENTRY glFramebufferTextureEXTContextANGLE(GLeglContext ctx,
                                                      GLenum target,
                                                      GLenum attachment,
@@ -5330,6 +5689,11 @@
     return gl::GenSamplersContextANGLE(ctx, count, samplers);
 }
 
+void GL_APIENTRY glGenSemaphoresEXTContextANGLE(GLeglContext ctx, GLsizei n, GLuint *semaphores)
+{
+    return gl::GenSemaphoresEXTContextANGLE(ctx, n, semaphores);
+}
+
 void GL_APIENTRY glGenTexturesContextANGLE(GLeglContext ctx, GLsizei n, GLuint *textures)
 {
     return gl::GenTexturesContextANGLE(ctx, n, textures);
@@ -5638,6 +6002,14 @@
     return gl::GetMaterialxvContextANGLE(ctx, face, pname, params);
 }
 
+void GL_APIENTRY glGetMemoryObjectParameterivEXTContextANGLE(GLeglContext ctx,
+                                                             GLuint memoryObject,
+                                                             GLenum pname,
+                                                             GLint *params)
+{
+    return gl::GetMemoryObjectParameterivEXTContextANGLE(ctx, memoryObject, pname, params);
+}
+
 void GL_APIENTRY glGetMultisamplefvContextANGLE(GLeglContext ctx,
                                                 GLenum pname,
                                                 GLuint index,
@@ -5892,6 +6264,14 @@
     return gl::GetSamplerParameterivContextANGLE(ctx, sampler, pname, params);
 }
 
+void GL_APIENTRY glGetSemaphoreParameterui64vEXTContextANGLE(GLeglContext ctx,
+                                                             GLuint semaphore,
+                                                             GLenum pname,
+                                                             GLuint64 *params)
+{
+    return gl::GetSemaphoreParameterui64vEXTContextANGLE(ctx, semaphore, pname, params);
+}
+
 void GL_APIENTRY glGetShaderInfoLogContextANGLE(GLeglContext ctx,
                                                 GLuint shader,
                                                 GLsizei bufSize,
@@ -6124,6 +6504,19 @@
     return gl::GetUniformuivContextANGLE(ctx, program, location, params);
 }
 
+void GL_APIENTRY glGetUnsignedBytevEXTContextANGLE(GLeglContext ctx, GLenum pname, GLubyte *data)
+{
+    return gl::GetUnsignedBytevEXTContextANGLE(ctx, pname, data);
+}
+
+void GL_APIENTRY glGetUnsignedBytei_vEXTContextANGLE(GLeglContext ctx,
+                                                     GLenum target,
+                                                     GLuint index,
+                                                     GLubyte *data)
+{
+    return gl::GetUnsignedBytei_vEXTContextANGLE(ctx, target, index, data);
+}
+
 void GL_APIENTRY glGetVertexAttribIivContextANGLE(GLeglContext ctx,
                                                   GLuint index,
                                                   GLenum pname,
@@ -6187,6 +6580,23 @@
     return gl::HintContextANGLE(ctx, target, mode);
 }
 
+void GL_APIENTRY glImportMemoryFdEXTContextANGLE(GLeglContext ctx,
+                                                 GLuint memory,
+                                                 GLuint64 size,
+                                                 GLenum handleType,
+                                                 GLint fd)
+{
+    return gl::ImportMemoryFdEXTContextANGLE(ctx, memory, size, handleType, fd);
+}
+
+void GL_APIENTRY glImportSemaphoreFdEXTContextANGLE(GLeglContext ctx,
+                                                    GLuint semaphore,
+                                                    GLenum handleType,
+                                                    GLint fd)
+{
+    return gl::ImportSemaphoreFdEXTContextANGLE(ctx, semaphore, handleType, fd);
+}
+
 void GL_APIENTRY glInsertEventMarkerEXTContextANGLE(GLeglContext ctx,
                                                     GLsizei length,
                                                     const GLchar *marker)
@@ -6240,6 +6650,11 @@
     return gl::IsFramebufferOESContextANGLE(ctx, framebuffer);
 }
 
+GLboolean GL_APIENTRY glIsMemoryObjectEXTContextANGLE(GLeglContext ctx, GLuint memoryObject)
+{
+    return gl::IsMemoryObjectEXTContextANGLE(ctx, memoryObject);
+}
+
 GLboolean GL_APIENTRY glIsProgramContextANGLE(GLeglContext ctx, GLuint program)
 {
     return gl::IsProgramContextANGLE(ctx, program);
@@ -6270,6 +6685,11 @@
     return gl::IsRenderbufferOESContextANGLE(ctx, renderbuffer);
 }
 
+GLboolean GL_APIENTRY glIsSemaphoreEXTContextANGLE(GLeglContext ctx, GLuint semaphore)
+{
+    return gl::IsSemaphoreEXTContextANGLE(ctx, semaphore);
+}
+
 GLboolean GL_APIENTRY glIsSamplerContextANGLE(GLeglContext ctx, GLuint sampler)
 {
     return gl::IsSamplerContextANGLE(ctx, sampler);
@@ -6469,6 +6889,14 @@
     return gl::MemoryBarrierByRegionContextANGLE(ctx, barriers);
 }
 
+void GL_APIENTRY glMemoryObjectParameterivEXTContextANGLE(GLeglContext ctx,
+                                                          GLuint memoryObject,
+                                                          GLenum pname,
+                                                          const GLint *params)
+{
+    return gl::MemoryObjectParameterivEXTContextANGLE(ctx, memoryObject, pname, params);
+}
+
 void GL_APIENTRY glMultMatrixfContextANGLE(GLeglContext ctx, const GLfloat *m)
 {
     return gl::MultMatrixfContextANGLE(ctx, m);
@@ -7178,6 +7606,14 @@
     return gl::ScissorContextANGLE(ctx, x, y, width, height);
 }
 
+void GL_APIENTRY glSemaphoreParameterui64vEXTContextANGLE(GLeglContext ctx,
+                                                          GLuint semaphore,
+                                                          GLenum pname,
+                                                          const GLuint64 *params)
+{
+    return gl::SemaphoreParameterui64vEXTContextANGLE(ctx, semaphore, pname, params);
+}
+
 void GL_APIENTRY glSetFenceNVContextANGLE(GLeglContext ctx, GLuint fence, GLenum condition)
 {
     return gl::SetFenceNVContextANGLE(ctx, fence, condition);
@@ -7207,6 +7643,18 @@
     return gl::ShaderSourceContextANGLE(ctx, shader, count, string, length);
 }
 
+void GL_APIENTRY glSignalSemaphoreEXTContextANGLE(GLeglContext ctx,
+                                                  GLuint semaphore,
+                                                  GLuint numBufferBarriers,
+                                                  const GLuint *buffers,
+                                                  GLuint numTextureBarriers,
+                                                  const GLuint *textures,
+                                                  const GLenum *dstLayouts)
+{
+    return gl::SignalSemaphoreEXTContextANGLE(ctx, semaphore, numBufferBarriers, buffers,
+                                              numTextureBarriers, textures, dstLayouts);
+}
+
 void GL_APIENTRY glStencilFuncContextANGLE(GLeglContext ctx, GLenum func, GLint ref, GLuint mask)
 {
     return gl::StencilFuncContextANGLE(ctx, func, ref, mask);
@@ -7374,6 +7822,22 @@
                                       border, format, type, pixels);
 }
 
+void GL_APIENTRY glTexImage3DOESContextANGLE(GLeglContext ctx,
+                                             GLenum target,
+                                             GLint level,
+                                             GLenum internalformat,
+                                             GLsizei width,
+                                             GLsizei height,
+                                             GLsizei depth,
+                                             GLint border,
+                                             GLenum format,
+                                             GLenum type,
+                                             const void *pixels)
+{
+    return gl::TexImage3DOESContextANGLE(ctx, target, level, internalformat, width, height, depth,
+                                         border, format, type, pixels);
+}
+
 void GL_APIENTRY glTexParameterIivOESContextANGLE(GLeglContext ctx,
                                                   GLenum target,
                                                   GLenum pname,
@@ -7515,6 +7979,63 @@
                                                       height, depth, fixedsamplelocations);
 }
 
+void GL_APIENTRY glTexStorageMem2DEXTContextANGLE(GLeglContext ctx,
+                                                  GLenum target,
+                                                  GLsizei levels,
+                                                  GLenum internalFormat,
+                                                  GLsizei width,
+                                                  GLsizei height,
+                                                  GLuint memory,
+                                                  GLuint64 offset)
+{
+    return gl::TexStorageMem2DEXTContextANGLE(ctx, target, levels, internalFormat, width, height,
+                                              memory, offset);
+}
+
+void GL_APIENTRY glTexStorageMem2DMultisampleEXTContextANGLE(GLeglContext ctx,
+                                                             GLenum target,
+                                                             GLsizei samples,
+                                                             GLenum internalFormat,
+                                                             GLsizei width,
+                                                             GLsizei height,
+                                                             GLboolean fixedSampleLocations,
+                                                             GLuint memory,
+                                                             GLuint64 offset)
+{
+    return gl::TexStorageMem2DMultisampleEXTContextANGLE(
+        ctx, target, samples, internalFormat, width, height, fixedSampleLocations, memory, offset);
+}
+
+void GL_APIENTRY glTexStorageMem3DEXTContextANGLE(GLeglContext ctx,
+                                                  GLenum target,
+                                                  GLsizei levels,
+                                                  GLenum internalFormat,
+                                                  GLsizei width,
+                                                  GLsizei height,
+                                                  GLsizei depth,
+                                                  GLuint memory,
+                                                  GLuint64 offset)
+{
+    return gl::TexStorageMem3DEXTContextANGLE(ctx, target, levels, internalFormat, width, height,
+                                              depth, memory, offset);
+}
+
+void GL_APIENTRY glTexStorageMem3DMultisampleEXTContextANGLE(GLeglContext ctx,
+                                                             GLenum target,
+                                                             GLsizei samples,
+                                                             GLenum internalFormat,
+                                                             GLsizei width,
+                                                             GLsizei height,
+                                                             GLsizei depth,
+                                                             GLboolean fixedSampleLocations,
+                                                             GLuint memory,
+                                                             GLuint64 offset)
+{
+    return gl::TexStorageMem3DMultisampleEXTContextANGLE(ctx, target, samples, internalFormat,
+                                                         width, height, depth, fixedSampleLocations,
+                                                         memory, offset);
+}
+
 void GL_APIENTRY glTexSubImage2DContextANGLE(GLeglContext ctx,
                                              GLenum target,
                                              GLint level,
@@ -7547,6 +8068,23 @@
                                          height, depth, format, type, pixels);
 }
 
+void GL_APIENTRY glTexSubImage3DOESContextANGLE(GLeglContext ctx,
+                                                GLenum target,
+                                                GLint level,
+                                                GLint xoffset,
+                                                GLint yoffset,
+                                                GLint zoffset,
+                                                GLsizei width,
+                                                GLsizei height,
+                                                GLsizei depth,
+                                                GLenum format,
+                                                GLenum type,
+                                                const void *pixels)
+{
+    return gl::TexSubImage3DOESContextANGLE(ctx, target, level, xoffset, yoffset, zoffset, width,
+                                            height, depth, format, type, pixels);
+}
+
 void GL_APIENTRY glTransformFeedbackVaryingsContextANGLE(GLeglContext ctx,
                                                          GLuint program,
                                                          GLsizei count,
@@ -8020,6 +8558,18 @@
     return gl::ViewportContextANGLE(ctx, x, y, width, height);
 }
 
+void GL_APIENTRY glWaitSemaphoreEXTContextANGLE(GLeglContext ctx,
+                                                GLuint semaphore,
+                                                GLuint numBufferBarriers,
+                                                const GLuint *buffers,
+                                                GLuint numTextureBarriers,
+                                                const GLuint *textures,
+                                                const GLenum *srcLayouts)
+{
+    return gl::WaitSemaphoreEXTContextANGLE(ctx, semaphore, numBufferBarriers, buffers,
+                                            numTextureBarriers, textures, srcLayouts);
+}
+
 void GL_APIENTRY glWaitSyncContextANGLE(GLeglContext ctx,
                                         GLsync sync,
                                         GLbitfield flags,
@@ -9198,4 +9748,28 @@
     return gl::ProvokingVertexANGLEContextANGLE(ctx, mode);
 }
 
+void GL_APIENTRY glLoseContextCHROMIUMContextANGLE(GLeglContext ctx, GLenum current, GLenum other)
+{
+    return gl::LoseContextCHROMIUMContextANGLE(ctx, current, other);
+}
+
+void GL_APIENTRY glTexImage2DExternalANGLEContextANGLE(GLeglContext ctx,
+                                                       GLenum target,
+                                                       GLint level,
+                                                       GLint internalformat,
+                                                       GLsizei width,
+                                                       GLsizei height,
+                                                       GLint border,
+                                                       GLenum format,
+                                                       GLenum type)
+{
+    return gl::TexImage2DExternalANGLEContextANGLE(ctx, target, level, internalformat, width,
+                                                   height, border, format, type);
+}
+
+void GL_APIENTRY glInvalidateTextureANGLEContextANGLE(GLeglContext ctx, GLenum target)
+{
+    return gl::InvalidateTextureANGLEContextANGLE(ctx, target);
+}
+
 }  // extern "C"